### PR TITLE
Fixing bugs with TTL updates

### PR DESCRIPTION
**Pull request by vgkholla**
1. Corner case with PUT, TTL update and journal
   If the TTL update is in the journal and the PUT associated with
   it is in a different index segment, there would have been a NPE
2. Add Blob_Update_Not_Allowed to the list of non system health
   errors at the Router

`./gradlew build && ./gradlew test` succeeded

### CODE REVIEW

File: ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java [hunk 1]
```diff
@@ -1288,7 +1288,7 @@ private void updateStateForMessages(List<MessageInfo> messageEntries) throws Sto
       MessageInfo messageInfo = messageEntriesIterator.next();
       if (!messageInfo.isDeleted()) {
         // ok to use most recent ref
-        IndexValue indexValue = findKey(messageInfo.getStoreKey());
+        IndexValue indexValue = findKey(messageInfo.getStoreKey(), null, EnumSet.allOf(IndexEntryType.class));
```
[the remark this answers is not in the document]
Comment (vgkholla) [reply ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java]: Done

[the remark this answers is not in the document]
Comment (vgkholla) [reply ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java]: Done

**Comment by vgkholla** [comment]
@pnarayanan comment addressed !

**Comment by zzmao** [comment]
You may need to update the comments after calling `updateStateForMessages()`
```logger.trace("Segment based token, Time used to update delete state: {}",```

File: ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java [hunk 2]
```diff
@@ -509,6 +509,9 @@ private boolean needToRollOverIndex(IndexEntry entry) {
   /**
    * Finds a key in the index and returns the blob index value associated with it. If not found,
    * returns null
+   * <p/>
```
Comment (pnarayanan) [inline ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java]: wrongly formatted?

Comment (vgkholla) [reply ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java]: I don't think so but since github is flagging it, I'll just remove tit

Comment (vgkholla) [reply ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java]: Actually I found other places that use this - so should be ok

**Comment by vgkholla** [comment]
@zzmao Fixed it !

**Comment by vgkholla** [comment]
@pnarayanan @zzmao `./gradlew build && ./gradlew test` succeeded

**Pull request merged**